packages/kbn-es/src/cluster.js

```diff
@@ -306,12 +306,28 @@ exports.Cluster = class Cluster {
     this._log.info(chalk.bold('Starting'));
     this._log.indent(4);
 
-    const esArgs = [
+    let defaultEsArgs = [
       'action.destructive_requires_name=true',
-      'ingest.geoip.downloader.enabled=false',
-      'search.check_ccs_compatibility=true',
       'cluster.routing.allocation.disk.threshold_enabled=false',
-    ].concat(options.esArgs || []);
+    ];
+
+    if (process.env.JEST_WORKER_ID || opts.testDefaults) {
+      defaultEsArgs = defaultEsArgs.concat([
+        'ingest.geoip.downloader.enabled=false',
+        'search.check_ccs_compatibility=true',
+      ]);
+    }
```

Comment:
Our goals with these defaults changes were to provide Kibana developers an ES instance that has the same limitations we impose on instances in CI, and I don't think moving these default to be opt-in is a good idea. The geoid.downloader makes ES startup time significantly worse (which piles up to a lot of extra time on CI) and disabling check_ccs_compatibility removes a safeguard that I don't think most contributors understand they need to opt-into. I think we should keep these as the defaults, but also add the ability to change the default specified values as you've done below.

Reply:
Yeah I reckon this part was controversial 😄 

Would you be open to flipping the flag so it's easy to opt out of all the kibana defaults / opt in to run with Elasticsearch defaults?

Reply:
I'd much rather require that people turning off ccs compatibility checks mention it explicitly. I don't want it to become another `--no-base-path` situation where a handful of influential people tell everyone they work with to use it and they don't know what they're doing.

Reply:
Cool I can appreciate and relate those concerns will amend the PR accordingly.